### PR TITLE
MDS: switch to safe mode when close to convergence

### PR DESCRIPTION
**Pull request by nychiang**
When speculative mode is selected with MDS system, hiop starts from fast mode, and switch to stable mode when approaching convergence.

### CODE REVIEW

File: src/Optimization/hiopAlgFilterIPM.cpp [hunk 1]
```diff
@@ -1511,6 +1511,33 @@ hiopAlgFilterIPMNewton::switch_to_safer_KKT(hiopKKTLinSys* kkt_curr,
     } // end of if(kkt)
   }
 #endif
+
+// TODO: turn to 0 --- keep using fast mode till linear solver (kkt->update) fails
+#if 1
```
Comment (nychiang) [inline src/Optimization/hiopAlgFilterIPM.cpp:1516]: @cnpetra both work fine with all the unit tests. Will test these two options with exago test cases. Please let me know if you have preference.

Comment (cnpetra) [reply src/Optimization/hiopAlgFilterIPM.cpp:1516]: > @cnpetra both work fine with all the unit tests. Will test these two options with exago test cases. Please let me know if you have preference.

looks good. I have a couple of questions/remarks for when we meet in person.

**Review by cnpetra — APPROVED** [review]

**Comment by cnpetra** [comment]
@nychiang can you resolve conflicts?

**Comment by nychiang** [comment]
> @nychiang can you resolve conflicts?

Just rebased to the latest `develop` branch

**Pull request merged**